lib/json_api_client/resource.rb

```diff
@@ -348,6 +350,17 @@ def _build_connection(rebuild = false)
           yield(conn) if block_given?
         end
       end
+
+      def _name_for_route_format(name)
```

Comment:
Can this be a method on `name`? like `name.to_route(route_format)`?

Reply:
@gaorlov name here is a Symbol, that would require a big refactor in other places as well and I believe that to be out of scope for this PR. But happy to discuss further to continue contributing 🙏

Reply:
Could you do like
```ruby
module Routable
  def to_route(format: :dasherized_route)
    case format
    when :dasherized_route
      to_s.dasherize
    when :camelized_route
      to_s.camelize(:lower)
    else
      self
    end
  end
end
 
class Symbol
  include Routable
end
```
and then you can have `name.to_route(route_format)` and not have to do anything else to the code